internal/device/recovery.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2023 The Armored Witness authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package device contains functions for dealing with the armored witness
+// hardware device.
+package device
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	"k8s.io/klog/v2"
+)
+
+// BootIntoRecovery attempts to boot an armored witness device in SDP (Serial Boot) mode
+// with the recovery firmware image, and then watches for a matching new block device
+// to be presented to the host.
+//
+// Returns the HID device and detected block device path.
+func BootIntoRecovery(ctx context.Context, recoveryFirmware []byte, blockDeviceGlob string) (*Target, string, error) {
```

Comment:
If I only saw the function description for this I would assume it would be more active and "do it or fail fast". Specifically, I wouldn't have expected a wait on the context to be the first thing.

Consider `AwaitDeviceAndBootIntoRecovery`, or just line in the docstring making it clear that the context must have a timeout if you don't want this to hang forever if no device appears in the correct state.

Reply:
I've updated the comment on the function to make it clearer, ta.